code/datums/components/armor_plate.dm

```diff
@@ -12,6 +12,7 @@
 	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
 	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/applyplate)
 	RegisterSignal(parent, COMSIG_PARENT_PREQDELETED, .proc/dropplates)
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_OVERLAYS, .proc/apply_overlays)
```

Comment:
Check if it's a ripley before registering and you don't need to istype in the receiving proc